gammapy/estimators/points/core.py

```diff
@@ -612,3 +614,41 @@ def plot_ts_profiles(
             ax.figure.colorbar(caxes, ax=ax, label=label)
 
         return ax
+
+    def recompute_ul(self, n_sigma_ul=2, **kwargs):
+        """Recompute upper limits corresponding to the given value.
+        The pre-computed stat profiles must exist for the re-computation.
+
+        Parameters
+        ----------
+        n_sigma_ul : int
+            Number of sigma to use for upper limit computation. Default is 2.
+        **kwargs : dict
+            Keyword arguments passed to `~scipy.optimize.brentq`.
+
+        Returns
+        --------
+        flux_points : `FluxPoints`
+            A new FluxPoints object with modified upper limits
+        """
+
+        if self.has_stat_profiles is False:
+            raise ValueError(
+                "Stat profiles not present. Upper limit computation is not possible"
+            )
+
+        delta_ts = n_sigma_ul ** 2
+
+        flux_points = copy(self)
+
+        value_scan = self.stat_scan.geom.axes["norm"].center
+        shape_axes = self.stat_scan.geom._shape[slice(3, None)]
+        for idx in np.ndindex(shape_axes):
```

Comment:
You could use `self.stat_scan.iter_by_axis()` (see https://docs.gammapy.org/0.19/api/gammapy.maps.RegionNDMap.html#gammapy.maps.RegionNDMap.iter_by_axis)

Reply:
I could not manage to simplify it using this while iterating over both `stat_scan` and `stat` with the generators...

Reply:
OK